### PR TITLE
Allow choosing "Edit" from any beatmap carousel item

### PR DESCRIPTION
**Pull request by peppy**
Was a bit confusing until now. Even I hit this at least once.

https://twitter.com/ppy/status/1736963119828897883

CleanShot 2023-12-19 at 11 00 48

### CODE REVIEW

File: osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs [hunk 1]
```diff
@@ -222,6 +227,9 @@ public MenuItem[] ContextMenuItems
                 if (Item?.State.Value == CarouselItemState.NotSelected)
                     items.Add(new OsuMenuItem("Expand", MenuItemType.Highlighted, () => Item.State.Value = CarouselItemState.Selected));
 
+                if (mainMenuItems != null)
```
Comment (sharkmu) [inline osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs:230]: I found a simplified way to do this:
```cs
items.AddRange(mainMenuItems ?? Array.Empty<MenuItem>());
```
What do you think?

Comment (bdach) [reply osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs:230]: nope

File: osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs [hunk 2]
```diff
@@ -57,8 +59,11 @@ protected override void FreeAfterUse()
         }
 
         [BackgroundDependencyLoader]
-        private void load(BeatmapSetOverlay? beatmapOverlay)
+        private void load(BeatmapSetOverlay? beatmapOverlay, SongSelect? songSelect)
         {
+            if (songSelect != null)
+                mainMenuItems = songSelect.CreateForwardNavigationMenuItemsForBeatmap(() => (((CarouselBeatmapSet)Item!).GetNextToSelect() as CarouselBeatmap)!.BeatmapInfo);
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs:65]: > I was going to suggest only permitting this if the beatmap set panel is expanded, but that is also weirdly arbitrary...?

Yeah I also considered doing this.

And I get that it's weird. I'd hope that the future song select redesign removes any kind of confusion around carousel items, but we'll see...

I'm open to any other suggestion that helps get the "Edit" option visible in this panel's context menu.

[the remark this answers is not in the document]
Comment (bdach) [reply osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs:65]: I'm just gonna get this in as is for now ¯\\\_(ツ)\_/¯

**Review by bdach — APPROVED** [review]

**Pull request merged**